### ISSUE TITLE
Question: Can we use PyPI lm-eval instead of the arubique fork, and is stnd needed?

### ISSUE THREAD

**Issue by cemde** [parameterlab/MASEval#54]
## Context

For the v0.4.0 release, PyPI rejected the package because of direct git URL dependencies in `pyproject.toml`:

```
lm-eval @ git+https://github.com/arubique/lm-evaluation-harness.git@main
stnd @ git+https://github.com/arubique/stnd.git@0d23b52f7742c08b28be560d2d52d450fcd274b7
```

PyPI does not allow direct URL dependencies in published packages.

## Changes made

We replaced these with the official PyPI `lm-eval>=0.4.0` and removed `stnd` entirely. Here's the reasoning:

### `lm-eval`

The [arubique fork](https://github.com/arubique/lm-evaluation-harness) adds:
- Logit extraction during `generate_until()` (new `Instance.logits` field)
- `--use_full_prompt` CLI flag
- Fewshot sampling fixes
- Custom task variants (mmlu_prompts, anatomy, hellaswag_prompts, mmlu_pro_prompts)

However, maseval only imports `HFLM` and `Instance` for the `loglikelihood()` code path (in `precompute_all_logprobs_lmeval()`), which is **not modified** in the fork. The official PyPI version should work for maseval's library usage.

**Question for @arubique:** Are there any DISCO or example workflows that rely on the fork-specific features (custom tasks, logit extraction, `--use_full_prompt`)? If so, we may need to document a separate manual install step for those use cases.

### `stnd`

[`stnd`](https://github.com/arubique/stnd) is not published on PyPI and is **never imported anywhere** in the maseval codebase. It appears to be a phantom dependency — declared in the `disco` extras but unused. We removed it.

**Question for @arubique:** Is `stnd` needed for a DISCO workflow that isn't checked in yet, or is it safe to remove? If it's still needed for some use case, should we document it as a manual install step instead?

## Summary

| Dependency | Before | After | Rationale |
|---|---|---|---|
| `lm-eval` | `git+https://github.com/arubique/...` | `lm-eval>=0.4.0` (PyPI) | maseval only uses unmodified upstream APIs |
| `stnd` | `git+https://github.com/arubique/...` | removed | never imported in codebase |

**Comment by arubique** [comment]
Hi @cemde 

1. there is no code in MASEval that relies on the lm-eval's fork-specific features. We should change the dependency to the upstream lm-eval API.
2. stnd is not needed for the DISCO workflow. It is safe to remove.

Thanks for noticing these points!

I will open the corresponding PR.

**Comment by cemde** [comment]
Thanks @arubique I was 90% sure so I went ahead and already fixed it, just wanted to check. No action needed.
